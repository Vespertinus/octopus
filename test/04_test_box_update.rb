class UpdateEnv < RunEnv
  def config
    connect_string, config = super
    config += <<EOD
object_space[0].enabled = 1
object_space[0].index[0].type = "HASH"
object_space[0].index[0].unique = 1
object_space[0].index[0].key_field[0].fieldno = 0
object_space[0].index[0].key_field[0].type = "STR"
EOD
    return connect_string, config
  end
end

UpdateEnv.new.with_server do |box|
  box.ping
  box.insert [1, 2, "abc", "def"]

  box.update_fields 1, [1, :or, 0xff]
  box.select 1

  box.update_fields 1, [1, :set, 127]
  box.select 1

  box.update_fields 1, [2, :splice, 2, 4, "aaaa"]
  box.select 1

  box.update_fields 1, [2, :splice, 2, 4, ""]
  box.select 1

  box.update_fields 1, [1, :insert, 11]
  box.select 1

  box.update_fields 1, [1, :delete, ""]
  box.select 1

  box.update_fields 1, [1, :delete, ""], [1, :delete, ""], [1, :delete, ""]
  box.select 1

  LogPpProxy.try { box.update_fields 1, [1, :delete, ""] }
  box.select 1

  box.insert [1]
  box.update_fields 1, [1, :insert, "aa"]
  box.select 1
end
