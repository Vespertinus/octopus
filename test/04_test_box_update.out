# box.ping()
:pong

# box.insert([1, 2, "abc", "def"])
1

# box.update_fields(1, [1, :or, 255])
1

# box.select(1)
[["\001\000\000\000", "\375\000\000\000", "abc", "def"]]

# box.update_fields(1, [1, :set, 127])
1

# box.select(1)
[["\001\000\000\000", "\177\000\000\000", "abc", "def"]]

# box.update_fields(1, [2, :splice, 2, 4, "aaaa"])
1

# box.select(1)
[["\001\000\000\000", "\177\000\000\000", "abaaaa", "def"]]

# box.update_fields(1, [2, :splice, 2, 4, ""])
1

# box.select(1)
[["\001\000\000\000", "\177\000\000\000", "ab", "def"]]

# box.update_fields(1, [1, :insert, 11])
1

# box.select(1)
[["\001\000\000\000", "\v\000\000\000", "\177\000\000\000", "ab", "def"]]

# box.update_fields(1, [1, :delete, ""])
1

# box.select(1)
[["\001\000\000\000", "\177\000\000\000", "ab", "def"]]

# box.update_fields(1, [1, :delete, ""], [1, :delete, ""], [1, :delete, ""])
1

# box.select(1)
[["\001\000\000\000"]]

# box.update_fields(1, [1, :delete, ""])
Failed with: {code: 0x202, message: 'update of field beyond tuple cardinality'}
# box.select(1)
[["\001\000\000\000"]]

# box.insert([1])
2

# box.update_fields(1, [1, :insert, "aa"])
1

# box.select(1)
[["\001\000\000\000", "aa"]]

# box.insert([10, "foo"])
1

# box.update_fields(10, [1, :splice, 0, 1, "b"])
1

# box.select(10)
[["\n\000\000\000", "boo"]]

# box.insert([10, "foo"])
2

# box.update_fields(10, [1, :set, ""])
1

# box.update_fields(10, [1, :set, ""])
1

# box.select(10)
[["\n\000\000\000", ""]]

# box.insert([2, "foo"])
1

# box.insert([3, "bar"])
1

# box.select(2, 3, 4)
[["\002\000\000\000", "foo"], ["\003\000\000\000", "bar"]]

# box.update_fields(2, [0, :set, 4])
1

# box.select(2, 3, 4)
[["\003\000\000\000", "bar"], ["\004\000\000\000", "foo"]]

# box.update_fields(4, [0, :set, 3])
Failed with: {code: 0x3802, message: 'duplicate key value violates unique index 0:StringHash'}
